velox/exec/ExchangeQueue.cpp

```diff
@@ -86,7 +96,13 @@ void ExchangeQueue::enqueueLocked(
   receivedBytes_ += page->size();
 
   queue_.push_back(std::move(page));
-  if (!promises_.empty()) {
+  const auto minBatchSize = minOutputBatchBytesLocked();
+  while (!promises_.empty()) {
+    const auto unblockedConsumers = numberOfConsumers_ - promises_.size();
+    const auto unasignedBytes = totalBytes_ - unblockedConsumers * minBatchSize;
```

Comment:
Please make sure unasignedBytes is signed integer?

Reply:
Great catch. It is unsigned. Let me set the type explicitly.